internal/tls/tls.go

```diff
@@ -67,6 +67,10 @@ type credentials struct {
 	sn         string
 	insecure   bool
 	clientAuth tls.ClientAuthType
+	// hotReload toggles per-handshake reload using GetCertificate.
+	hotReload bool
+	// store keeps the latest loaded certificate.
+	store *certStore
```

Comment:
Since the certStore is essentially unused elsewhere, there is no need for store.go and certStore implementation. It would be simpler to just use the atomic.Pointer[*tls.Certificate] field here.